EpsilonAuxBMS/Src/main.c

```diff
@@ -505,17 +499,18 @@ static void MX_GPIO_Init(void)
 // Reimplement weak definition in stm32f4xx_hal_can.c
 void HAL_CAN_RxCpltCallback(CAN_HandleTypeDef* hcan)
 {
-    if (osMutexWait(orionStatus.orionStatusMutex, 0) != osOK)
-    {
-        return;
-    }
-
     CanRxMsgTypeDef* msg = hcan->pRxMsg;
 
-    if (msg->StdId == ORION_MAX_MIN_VOLTAGES_STDID)
+    if (msg->StdId == ORION_MAX_MIN_VOLTAGES_STDID && msg->DLC == 8)
     {
-        orionStatus.maxCellVoltage = (uint16_t)msg->Data[6]; // Max Cell voltage
-        orionStatus.minCellVoltage = (uint16_t)msg->Data[4]; // Min Cell Voltage
+        // Voltages are 2 bytes each, and memory is stored in little endian format
+        orionStatus.maxCellVoltage = (uint16_t)msg->Data[0] << 8 | msg->Data[1]; // Max Cell voltage
+        orionStatus.minCellVoltage = (uint16_t)msg->Data[2] << 8 | msg->Data[3]; // Min Cell Voltage
+    }
+    else if (msg->StdId == DRIVERS_INPUTS_STDID && msg->DLC == 4)
+    {
+        driversInput.aux = msg->Data[3] >> DRIVERS_INPUTS_AUX_BIT_POSITION & 1;
+        driversInput.forwardReverse = msg->Data[3] >> DRIVERS_INPUTS_FORWARD_REVERSE_BIT_POSITION & 1;
```

Comment:
Forward and Reverse are on two different bit indexes. Let's separate this variable into two.